internal/pkg/server/fleet.go

```diff
@@ -414,6 +421,12 @@ func (f *Fleet) runServer(ctx context.Context, cfg *config.Config) (err error) {
 		return err
 	}
 
+	go func() {
+		outputChanged := <-bulker.RemoteOutputCh()
+		f.outputCh <- outputChanged
```

Comment:
There is no for loop here so this can only ever receive one update before exiting, is that the intent?

I don't see this channel being closed anywhere and this goroutine doesn't select on a context so if there were a for loop this goroutine will never exit.

Reply:
If a remote output changed, fleet server will be restarted, do we need a for loop then?

Reply:
Restarting Fleet server takes all agents offline. I think it will almost always be the case that when multiple Fleet Severs are deployed for high availability that they will share a remote ES output.

This means that if the remote ES output goes offline, every Fleet server will restart will restart at once. I think it would be much better if changing the remote ES output did not require a restart. If we cannot do this, then rolling out the change gradually so as not to take every connected agent offline at once would be better.

Reply:
That said, I wouldn't expect the remote ES output to change frequently, so perhaps this is something to address as a follow up.

In general we should avoid anything that causes Fleet server to go offline as much as possible, even if it is only offline briefly.

At minimum if we keep this behavior, it needs to be very clear to users that if they change the remote ES configuration their Fleet servers are all going to restart. This is not something they will assume by default.

Reply:
I don't think we can ship with this behavior. It's going to blow away all the caches and create extra load on the main Elasticsearch cluster when a remote ES config is changed which is quite unexpected. Especially if you consider the use case where a very small management cluster is being used for Fleet + CCS and all data is sent to remote data clusters. Impact on these management clusters may be very high when the remote config is changed since the ES cluster may be sized just large enough to handle Fleet.

Is there a technical limitation to why we can't stop the old bulk and start a new one?

Reply:
Not really, I was trying to replicate the same behaviour as for local ES. If this is a concern, I can try to replace the bulker only if the remote es config changes.

Reply:
I'm a little concerned though about merging this pr eventually as it would go immediately to prod. Even if the remote ES output is not used yet, there might be changes impacting the existing functionality, which might not be caught by the smoke tests. 
I think we should run the elastic-agent e2e tests too as quality gates.

Reply:
It's not trivial to run elastic-agent e2e tests with a custom image, I might need agent team's help with it.